client/extensions/woocommerce/app/settings/email/mailchimp/mailchimp_dashboard.js

```diff
@@ -0,0 +1,249 @@
+	};
+
+	const checkbox = settings.mailchimp_checkbox_defaults;
+	const toggle = checkbox === 'check' || checkbox === 'uncheck';
+	return (
+		<div className="mailchimp__dashboard-settings">
+			<span className="mailchimp__dashboard-settings-form">
+				<FormFieldset>
+					<FormLegend>{ translate( 'Newsletter subscriptions' ) }</FormLegend>
+					<FormLabel>
+						<FormToggle
+							checked={ toggle }
+							onChange={ onToggle }
+							id="show-subscribe-message"
+						/>
+						<span>{ translate( 'Show a subscribe message to customer at checkout' ) }</span>
+					</FormLabel>
+					<FormLabel>
+						<FormRadio
+							disabled={ ! toggle }
+							value="check"
+							checked={ 'check' === checkbox }
+							onChange={ onRadioChange }
+						/>
+						<span>{ translate( 'Subscribe message is checked by default' ) }</span>
+					</FormLabel>
+					<FormLabel>
+						<FormRadio
+							disabled={ ! toggle }
+							value="uncheck"
+							checked={ 'uncheck' === checkbox }
+							onChange={ onRadioChange }
+						/>
+						<span>{ translate( 'Subscribe message is unchecked by default' ) }</span>
+					</FormLabel>
+					<FormLabel>
+						{ translate( 'Subscribe message' ) }
+					</FormLabel>
+					<FormTextInput
+						name="newsletter_label"
+						onChange={ onNewslatterLabelChange }
+						value={ settings.newsletter_label }
+					/>
+				</FormFieldset>
+			</span>
+			<span className="mailchimp__dashboard-settings-preview">
+				<div>{ translate( 'PREVIEW' ) }</div>
+				<div className="mailchimp__dashboard-settings-preview-view">
+					{ toggle && <FormLabel>
+							<FormCheckbox checked={ checkbox === 'check' } disabled={ true } />
+							<span>{ settings.newsletter_label }</span>
+						</FormLabel>}
+				</div>
+				<div className="mailchimp__dashboard-settings-save">
+					<Button
+						primary
+						onClick={ onSave }
+						busy={ isSubbmittingSettings }
+						disabled={ isSubbmittingSettings }>
+						{ translate( 'Save' ) }
+					</Button>
+				</div>
+			</span>
+		</div>
+	);
+} );
+
+class MailChimpDashboard extends React.Component {
+
+	constructor( props ) {
+		super( props );
+		this.state = {
+			syncStatus: null,
+			settings: props.settings,
+		};
+	}
+
+	componentWillReceiveProps( nextProps ) {
+		const { translate } = nextProps;
+		if ( ( false === nextProps.isSubmittingNewsletterSetting ) && this.props.isSubmittingNewsletterSetting ) {
+			if ( nextProps.newsletterSettingsSubmitError ) {
+				nextProps.errorNotice( translate( 'There was a problem saving the email settings. Please try again.' ) );
+			} else {
+				nextProps.successNotice( translate( 'Email settings saved.' ), { duration: 4000 } );
+			}
+		}
+	}
+
+	onSettingsChange = ( change ) => {
+		this.setState( { settings: Object.assign( {}, this.state.settings, change ) } );
+	}
+
+	onSave = () => {
+		const { submitMailchimpNewsletterSettings: submit, siteId } = this.props;
```

Comment:
Why this reassignment? Why not call the mapped dispatcher by its original name?

Reply:
name clash with import.